### PR TITLE
boards: arm: Add support for STM NUCLEO-H753ZI

### PR DESCRIPTION
**Pull request by JeremyBCD**
Based on NUCLEO-H743ZI. NUCLEO-H753ZI adds support for hardware crypto and secure boot.

Also added `CONFIG_SOC_STM32H753XX` to `drivers/clock_control/clock_stm32_ll_h7.c`.

### CODE REVIEW

**Comment by JeremyBCD** [comment]
I'm unable to resolve the style issue in the compliance check, the change matches the style in that file, but not the project style guide.

**Comment by JeremyBCD** [comment]
Fixed `boards/arm/nucleo_h753zi/support/openocd.cfg`, using `board/st_nucleo_h743zi.cfg`.

**Review by FRASTM — APPROVED** [review]
maybe "Copyright (c) 202**1**" ?

**Comment by JeremyBCD** [comment]
@FRASTM Good point—updated copyright.

File: boards/arm/nucleo_h753zi/Kconfig.board [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# STM32H753ZI Nucleo board configuration
+
+# Copyright (c) 2020 Teslabs Engineering S.L.
```
Comment (erwango) [inline boards/arm/nucleo_h753zi/Kconfig.board:3]: Please keep only your copyright

File: boards/arm/nucleo_h753zi/Kconfig.defconfig [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# STM32H753ZI Nucleo board configuration
+
+# Copyright (c) 2020 Teslabs Engineering S.L.
```
Comment (erwango) [inline boards/arm/nucleo_h753zi/Kconfig.defconfig:3]: Please keep only your copyright

File: boards/arm/nucleo_h753zi/arduino_r3_connector.dtsi [hunk 3]
```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2019 Linaro Limited
+ * Copyright (c) 2020 Teslabs Engineering S.L.
```
Comment (erwango) [inline boards/arm/nucleo_h753zi/arduino_r3_connector.dtsi:3]: Please keep only your copyright

**Review by erwango — CHANGES_REQUESTED** [review]

**Comment by JeremyBCD** [comment]
@erwango Thanks for the review—I've removed the extraneous copyright, and moved the test and driver changes into separate commits.

File: drivers/clock_control/clock_stm32_ll_h7.c [hunk 4]
```diff
@@ -1,7 +1,6 @@
 /*
  *
- * Copyright (c) 2019 Linaro Limited.
- * Copyright (c) 2020 Jeremy LOCHE
+ * Copyright (c) 2021 Blue Clover Devices
```
Comment (erwango) [inline drivers/clock_control/clock_stm32_ll_h7.c:3]: I might be the one to blame here:
Please keep existing copyrights here (you're not adding a new file)

**Review by erwango — CHANGES_REQUESTED** [review]
Thanks for the changes.
One final tweak and we're good to go.

File: drivers/clock_control/clock_stm32_ll_h7.c (hunk 4 above)
Comment (JeremyBCD) [reply drivers/clock_control/clock_stm32_ll_h7.c]: That's my bad, I should have realized that from the contribution guidelines.

Fixed here and in the adc tests.

**Review by erwango — APPROVED** [review]

**Pull request merged**